api/src/main/java/org/openmrs/api/impl/ProgramWorkflowServiceImpl.java

```diff
@@ -69,38 +69,57 @@ public void setProgramWorkflowDAO(ProgramWorkflowDAO dao) {
 	 */
 	@Override
 	public Program saveProgram(Program program) throws APIException {
-		// Program
+
 		if (program.getConcept() == null) {
 			throw new APIException("Program.concept.required", (Object[]) null);
 		}
-		
-		// ProgramWorkflow
+
 		for (ProgramWorkflow workflow : program.getAllWorkflows()) {
-			
-			if (workflow.getConcept() == null) {
-				throw new APIException("ProgramWorkflow.concept.required", (Object[]) null);
-			}
-			if (workflow.getProgram() == null) {
-				workflow.setProgram(program);
-			} else if (!workflow.getProgram().equals(program)) {
-				throw new APIException("Program.error.contains.ProgramWorkflow", new Object[] { workflow.getProgram() });
-			}
-			
-			// ProgramWorkflowState
+			ensureProgramIsSet(workflow, program);
+
 			for (ProgramWorkflowState state : workflow.getStates()) {
-				
-				if (state.getConcept() == null || state.getInitial() == null || state.getTerminal() == null) {
-					throw new APIException("ProgramWorkflowState.requires", (Object[]) null);
-				}
-				if (state.getProgramWorkflow() == null) {
-					state.setProgramWorkflow(workflow);
-				} else if (!state.getProgramWorkflow().equals(workflow)) {
-					throw new APIException("ProgramWorkflow.error.contains.state", new Object[] { workflow.getProgram() });
-				}
+				ensureProgramWorkflowIsSet(state, workflow);
 			}
 		}
 		return dao.saveProgram(program);
 	}
+
+	/**
+	 * Helper method to ensure the program is set to the given workflow by doing the validations
+	 * @param workflow
```

Comment:
@JudeNiroshan please remove your javadocs. They dont add value/information. The methods are short, dont do much, and have meaningful names. Its easy for the reader to see what they do. And most importantly these are private methods. If they were public then of course you need to add proper javadocs (but then you would also need to fill in the parameter doc as codacy told you)